test/units/modules/cloud/vmware/test_data/test_vmware_guest_with_parameters.json

```diff
@@ -0,0 +1,106 @@
+[
+    [
+        {
+            "name": "sample_vm_001",
+            "hostname": "esxi01.example.com",
+            "username": "administrator@vsphere.local",
+            "password": "Secret@123$%",
+            "validate_certs": "True"
+        },
+        {
+            "failed": "True",
+            "msg": "Unknown error while connecting to vCenter or ESXi API"
```

Comment:
My definition of unit testing is a test that isolates the module being tested.  This means attempting to isolate it as much as possible.  I'm not seeing how the interaction with pyVmomi is being mocked or patched.  Without patching pyVmomi responses, it won't be possible to unit test failure paths.

Reply:
Like I said this is just starting efforts, and will continue to add more testcases based on pyVmomi mocking.